views/dashboard/services/components/ServiceEventsWidget/servicesEventForm.tsx

```diff
@@ -0,0 +1,165 @@
+import { useEffect, useMemo } from 'react'
+import { enqueueSnackbar } from 'notistack'
+import { useSelector, useDispatch } from 'react-redux'
+
+import useForm from '@/utils/hooks/useForm'
+import FormValidations from '@/utils/validations/forms'
+import { addOfferingEvent } from '@/lib/db/cms/services'
+import { TextField, FileUpload, FormButtonGroup } from '@/components/form'
+import { handleCloudUpload } from '@/utils/fileConversions'
+import { FileItem, FilesList } from '@/components/form/fields/FileUpload'
+import { StyledFieldSet } from '@/components/drawer/content/StyledFormComponents'
+import { statuses, messages } from '@/utils/banners'
+import { StyledServicesEventform } from '../../StyledServicesWidget'
+import { setServices } from '@/lib/redux/slices/services'
+
+interface EventFormFields {
+    description?: string
+    img?: string
+    title?: string
+}
+
+interface ServicesEventFormTypes extends EventFormFields {
+    id: string
+    section: string
+    label: 'Adding' | 'Editing'
+    store: EventFormFields
+}
+
+const initialState = {
+    description: { value: '' },
+    title: { value: '' },
+    img: { value: [] }
+}
+
+const validations = {
+    description: FormValidations.yolo,
+    title: FormValidations.yolo,
+    img: FormValidations.yolo
+}
+
+const ServicesEventForm = ({
+    id,
+    section,
+    description,
+    img,
+    title,
+    label,
+    store
+}: ServicesEventFormTypes) => {
+    const dispatch = useDispatch()
+    const { id: serviceID } = useSelector((state: any) => state.services)
+
+    const renderLabel =
+        label === 'Adding'
+            ? `Adding a new event to ${section}`
+            : `Editing event for ${section}`
+
+    const {
+        handleChange,
+        handleNonFormEventChange,
+        form,
+        formLoading,
+        handleImport,
+        isDataImported,
+        handleFormSubmit,
+        handleReset
+    } = useForm(initialState, validations, store)
+
+    useEffect(() => {
+        const hasData = Object.values(store).some(item => !!item)
+        if (hasData && !isDataImported) {
+            handleImport(store)
+        }
+    }, [store, isDataImported])
+
+    const onFilesChange = (files: any) => {
+        handleNonFormEventChange(files, 'img')
+    }
+
+    const disableSubmit = useMemo(() => {
+        const formStatuses = Object.entries(form).map(
+            ([label, field]: [label: string, field: any]) => {
+                if (label === 'img') {
+                    return field.value?.length > 0
+                }
+
+                return !!field.value
+            }
+        )
+        return !formStatuses.every(item => item)
+    }, [form])
+
+    const onFormSubmit = async () => {
+        const payload = {} as any
```

Comment:
My boy pulled a bart and didnt reset the form on submit ? #feature

Reply:
#sad